bindings/python/htcondor2/_dagman.py

```diff
@@ -0,0 +1,194 @@
+from typing import Union
+from typing import Optional
+from typing import List
+from typing import Tuple
+
+from ._common_imports import classad
+from .htcondor2_impl import _send_generic_payload_command
+from ._schedd import Schedd
+
+import enum
+
+class DagmanConnectionError(Exception):
+    """
+    Custom error for internal use to be raised when a DAGMan
+    connection fails and should be retried one more time.
+
+    :meta private:
+    """
+    def __init__(self, msg: Optional[str] = None) -> None:
+        self.message = msg if msg is not None else "Failed to connect to DAGMan"
+        super().__init__(self.message)
+
+
+class DAGMan():
+    """
+    The DAGMan client that acts as a connection to a running DAGMan
+    process for sending commands.
+    """
+
+    # Actual DaemonCore command int from condor_commands.h
+    CMD_DAGMAN_GENERIC = 61500
+
+    # Keep inline with dagman_commands.h enumeration
+    class Command(enum.IntEnum):
+        """
+        Internal mapping of specific generic payload DAG commands.
+
+        :meta private:
+        """
+
+        HALT = 1
+        RESUME = 2
+
+    class ContactInfo():
+        """
+        DAGMan contact information. Sinful address to send command
+        and secret DAGMan will use to verify trust.
+
+        :meta private:
+        """
+        def __init__(self, addr: str, secret: str) -> None:
+            self.addr = addr
+            self.secret = secret
+
+
+    def __init__(self, dag_id: Optional[int] = None) -> None:
+        """
+        :param dag_id: The :ad-attr:`ClusterId` of a running DAG to
+                       locate once the first command is issued.
+        """
+        self.dag_id = dag_id
+        self.contact = None
+
+
+    def locate(self, dag_id: int) -> Tuple[int, Optional[str]]:
```

Comment:
Shouldn't the dag_id for locate() be optional, since it can be provided via __init__()?